QuantConnect.ExanteBrokerage/ExanteBrokerage.cs

```diff
@@ -0,0 +1,839 @@
+        /// </summary>
+        /// <remarks>This parameterless constructor is required for brokerages implementing <see cref="IDataQueueHandler"/></remarks>
+        public ExanteBrokerage()
+            : this(ExanteBrokerageOptions.FromConfig(),
+                Composer.Instance.GetPart<IDataAggregator>())
+        {
+        }
+
+        /// <summary>
+        /// Creates a new ExanteBrokerage
+        /// </summary>
+        /// <param name="options">Exante brokerage options. Required to create REST API client instance</param>
+        /// <param name="aggregator">consolidate ticks</param>
+        public ExanteBrokerage(
+            ExanteBrokerageOptions options,
+            IDataAggregator aggregator) : base("ExanteBrokerage")
+        {
+            _options = options;
+
+            Client = new ExanteClientWrapper(_options.ExanteClientOptions());
+
+            _aggregator = aggregator;
+            _subscriptionManager = new EventBasedDataQueueHandlerSubscriptionManager();
+            _subscriptionManager.SubscribeImpl += Subscribe;
+            _subscriptionManager.UnsubscribeImpl += (s, _) => Unsubscribe(s);
+
+            SymbolMapper = new ExanteSymbolMapper(Client, SupportedCryptoCurrencies);
+            _messageHandler = new BrokerageConcurrentMessageHandler<ExanteOrder>(OnUserMessage);
+
+            Client.StreamClient.GetOrdersStreamAsync(exanteOrder => { _messageHandler.HandleNewMessage(exanteOrder); });
+        }
+
+        #region IDataQueueHandler
+
+        /// <summary>
+        /// Subscribe to the specified configuration
+        /// </summary>
+        /// <param name="dataConfig">defines the parameters to subscribe to a data feed</param>
+        /// <param name="newDataAvailableHandler">handler to be fired on new data available</param>
+        /// <returns>The new enumerator for this subscription request</returns>
+        public IEnumerator<BaseData> Subscribe(SubscriptionDataConfig dataConfig, EventHandler newDataAvailableHandler)
+        {
+            if (!CanSubscribe(dataConfig.Symbol))
+            {
+                return Enumerable.Empty<BaseData>().GetEnumerator();
+            }
+
+            var enumerator = _aggregator.Add(dataConfig, newDataAvailableHandler);
+            _subscriptionManager.Subscribe(dataConfig);
+
+            return enumerator;
+        }
+
+        /// <summary>
+        /// Removes the specified configuration
+        /// </summary>
+        /// <param name="dataConfig">Subscription config to be removed</param>
+        public void Unsubscribe(SubscriptionDataConfig dataConfig)
+        {
+            _subscriptionManager.Unsubscribe(dataConfig);
+            _aggregator.Remove(dataConfig);
+        }
+
+        /// <summary>
+        /// Sets the job we're subscribing for
+        /// </summary>
+        /// <param name="job">Job we're subscribing for</param>
+        public void SetJob(LiveNodePacket job)
+        {
+        }
```

Comment:
Fixed by 7b3b56fafdf5c6063936e08d9923a4f3c7bdf206

Reply:
> new ExanteBrokerageFactory().CreateBrokerage(job, null);

Will create a new instance and current instance will not really be initialized? Can see examples at https://github.com/QuantConnect/Lean/pull/6047

Reply:
Why not initialised? Take a look at https://github.com/QuantConnect/Lean.Brokerages.Exante/blob/7b3b56fafdf5c6063936e08d9923a4f3c7bdf206/QuantConnect.ExanteBrokerage/ExanteBrokerageFactory.cs#L68-L95

It extracts all the necessary parameters from the `LiveNodePacket job`. I choose not to use `Initialise` method because it requires some fields to drop `readonly` modifier.

Another thing I noticed is that there is extra routine on connecting it. Since `ExanteClient` is REST HTTP, I decided drop the check. But pro forma if you ask I added it at 439f1e7ada21a023b198b6e418724a0ad90b31f2.

Reply:
> new ExanteBrokerageFactory().CreateBrokerage(job, null);

This code is from a method in an instance of an already created brokerage, but doing `new ExanteBrokerageFactory().CreateBrokerage(job, null)` you are creating a new factory and an entirely new brokerage instance here. Should follow examples at https://github.com/QuantConnect/Lean/pull/6047

Reply:
I've got your point. Fixed by f868937345c5d226925281813349981df6a11af8

Reply:
The current implementation doesn't seem to be following the pattern yet, for example `ExanteBrokerage()` vs `BitfinexBrokerage()` please review